pkg/idutil/imagewalker/imagewalker.go

```diff
@@ -60,12 +61,25 @@ func (w *ImageWalker) Walk(ctx context.Context, req string) (int, error) {
 	}
 
 	matchCount := len(images)
+	// to handle the `rmi -f` case where returned images are different but
+	// have the same short prefix.
+	identical := true
+	if matchCount > 1 {
+		for _, image := range images[1:] {
+			if image.Target.Digest != images[0].Target.Digest {
+				identical = false
+				break
+			}
+		}
```

Comment:
I think Walk() should just return the number of the unique images as an integer

Reply:
Do you mean instead of using `Identical`, use `UniqueImages` in `Found` (as in my new update). Since if we let `Walk` returns the number of unique images, the `OnFound` function(which does the delete) has been called within `Walk`